src/models/checkout-object/modified-checkout-object.ts

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable class-methods-use-this */
+import { EpochNumber, UidType } from '../../JSCommon/common-types';
+import { CheckoutObject } from './checkout-object';
+
+export const TABLE_NAME = 'CHECKOUT_DATA';
+
+/**
+ * item_id: The ID of the item being checked out.
+ * user_id: The ID of the user checking out the item.
+ * checkout_time: The time the item was checked out.
+ * return_time: The time the item was returned.
+ * hackathon: The hackathon that this checkout belongs to.
+ * firstname: First name of user linked to the user_id
+ * lastname: Last name of the user linked to the user_id
+ * 
+ */
+interface IModifiedCheckoutObjectApiModel {
+  item_id: number;
+  user_id: string;
+  checkout_time: EpochNumber;
+  return_time?: EpochNumber;
+  hackathon?: UidType;
+  firstname?: string;
```

Comment:
Why do we need first name and last name in this model. We can just join with user id.

Reply:
We are joining with userID. The result of the join includes this. This is a readonly object.